packages/core/src/components/TreeView/TreeViewNode.js

```diff
@@ -7,50 +7,131 @@ import TreeViewIcon from './TreeViewIcon';
 import TreeViewIndents from './TreeViewIndents';
 
 class TreeViewNode extends Component {
-  constructor(props) {
-    super(props);
-
-    // A node can be set to be expanded by default
-    this.state = {
-      expanded:
-        (props.node.hasOwnProperty('state') && props.node.state.expanded) ||
-        false
-    };
-
-    this.toggleExpand = this.toggleExpand.bind(this);
-    this.handleSelect = this.handleSelect.bind(this);
-  }
-
-  // Collapse the current node if any of its parents is collapsed. This should
-  // only fire for nodes that are level 2 or greater
-  componentWillReceiveProps(nextProps) {
+  static getDerivedStateFromProps(nextProps, prevState) {
+    // Collapse the current node if any of its parents is collapsed. This should
+    // only fire for nodes that are level 2 or greater
     if (!nextProps.visible && nextProps.level > 1) {
-      this.setState(() => ({ expanded: false }));
+      return { expanded: false };
+    }
+
+    // Roving tab index
+    const tabIndex =
+      nextProps.focusedNodeId === prevState.nodeId ||
+      (!nextProps.focusedNodeId && prevState.nodeId === '1-0')
```

Comment:
can you explain why nodeId === '1-0' is hardcoded in a comment? I don't quite follow this piece...